sdk/core/azure-core/inc/azure/core/logging/logging.hpp

```diff
@@ -9,143 +9,51 @@
-
-  namespace Details {
-    enum class Facility : uint16_t
-    {
-      Core = 1,
-      Storage = 100,
-    };
-
-    template <Facility> class LogClassificationProvider;
-
-    class LogClassificationsPrivate;
-  } // namespace Details
-
-  /**
-   * @brief Represents a set of log classifications.
-   */
-  class LogClassifications {
-    friend class Details::LogClassificationsPrivate;
-
-    std::set<LogClassification> m_classifications;
-    bool m_all;
-
-    explicit LogClassifications(bool all) : m_all(all) {}
-
-  public:
-    /**
-     * @brief Initialize the list of log classifications with `std::initializer_list`.
-     * @param list An initializer list.
-     */
-    LogClassifications(std::initializer_list<LogClassification> list)
-        : m_classifications(list), m_all(false)
-    {
-    }
-
-    /**
-     * @brief Initialize the list of log classifications with `std::set`.
-     * @param set A set of classifications.
-     */
-    explicit LogClassifications(std::set<LogClassification> set)
-        : m_classifications(std::move(set)), m_all(false)
-    {
-    }
-  };
-
-  /**
-   * @brief Represents a log classification.
+   * @param level Maximum log level.
    */
-  class LogClassification {
-    template <Details::Facility> friend class Details::LogClassificationProvider;
-    friend struct std::less<LogClassification>;
-
-    int32_t m_value;
-
-    constexpr explicit LogClassification(Details::Facility facility, int16_t number)
-        : m_value((static_cast<int32_t>(number) << 16) | static_cast<int32_t>(facility))
-    {
-    }
-
-    constexpr bool operator<(LogClassification const& other) const
-    {
-      return m_value < other.m_value;
-    }
-
-  public:
-    /**
-     * @brief Compare log classification to another one.
-     * @param other Another log classification to compare to.
-     * @return `true` if this log classification equals to \p other, `false` otherwise.
-     */
-    constexpr bool operator==(LogClassification const& other) const
-    {
-      return m_value == other.m_value;
-    }
-
-    /**
-     * @brief Compare log classification to another one.
-     * @param other Another log classification to compare to.
-     * @return `true` if this log classification does not equal to \p other, `false` otherwise.
-     */
-    constexpr bool operator!=(LogClassification const& other) const
-    {
-      return m_value != other.m_value;
-    }
-
-    /**
-     * @brief Represents a list of all classifications.
-     */
-    AZ_CORE_DLLEXPORT static LogClassifications const All;
-
-    /**
-     * @brief Represents an empty list of classifications.
-     */
-    AZ_CORE_DLLEXPORT static LogClassifications const None;
-  };
-
-  namespace Details {
-    template <Facility F> class LogClassificationProvider {
-    protected:
-      constexpr static auto Classification(int16_t number) { return LogClassification(F, number); }
-    };
-  } // namespace Details
+  void SetMaximumLogLevel(LogLevel level);
```

Comment:
Im' not sure that "maximum" is a good term here. 
Maybe just SetLogLevel and the docs explains what it means.